packages/now-python/src/index.ts

```diff
@@ -166,16 +199,39 @@ export const build = async ({
   // Use the system-installed version of `python3` when running via `vercel dev`
   const runtime = meta.isDev ? 'python3' : 'python3.6';
 
-  const globOptions: GlobOptions = {
-    cwd: workPath,
-    ignore:
-      config && typeof config.excludeFiles === 'string'
-        ? config.excludeFiles
-        : 'node_modules/**',
-  };
+  const outputFiles = await glob('**', workPath);
+
+  // Static analysis is impossible with Python. Instead, provide `includeFiles`
```

Comment:
That's not true--static analysis is possible with python and we would prefer to have that instead of include/exclude

Reply:
If this were true, it would be even better. I'm not aware of any solution though. The highly dynamic nature of Python makes it difficult to trace all required dependencies. E.g. `Pylint`, which is one of the most sophisticated linters for Python, is finding it difficult to detect, e.g., some TensorFlow or OpenCV imports.

What's your envisioned approach for static analysis? The only library that I could find is [`treeshaker`](https://github.com/sclabs/treeshaker), but it's a very young project and I'd expect edge cases where it may fail. I haven't tried it though.

I think it's valuable to have include/exclude as a manual fallback in any case. When static analysis comes at some point, great, but include/exclude solves a real problem right now.

Reply:
Python has the `ast` module which can analyze itself. https://docs.python.org/3/library/ast.html

I don't think we want to introduce `includeFiles` here because there is nothing additional to include. You're either including everything or you are excluding somethings.

Reply:
> Python has the ast module which can analyze itself. https://docs.python.org/3/library/ast.html

I'm quite sure it only works for static imports. For instance, [Pylint has issues detecting TensorFlow imports](https://github.com/PyCQA/pylint/issues/2603). Static analysis may be sugar for cases where it works, but I don't think it'll be as reliable as treeshaking is in Javascript.

> I don't think we want to introduce `includeFiles` here because there is nothing additional to include. You're either including everything or you are excluding somethings.
>
> Let's revert the `includeFiles` part of this PR.

I beg to differ. How would you express

```json
{
  "excludeFiles": "**",
  "includeFiles": "api/py-func-a/**"
}
```

without both? As mentioned above, negative ignore patterns don't work because of a bug in `node-glob`.

Reply:
Let's change the comment from `impossible` to `difficult`. I started work on a python equivalent to [@vercel/nft](https://github.com/vercel/nft) using `ast` and it it seems possible. Analyzing imported files is much easier than linting.

Reply:
> Let's change the comment from `impossible` to `difficult`.

Agreed, done.

> I started work on a python equivalent to @vercel/nft using `ast` and it it seems possible.

Sounds great, looking forward to seeing it in the Python runtime some time. Again, I don't have hands-on experience with static import analysis in Python, but, e.g., [Nuitka](https://github.com/Nuitka/Nuitka) (a Python compiler) seems to have some difficulties compiling packages with C extensions, e.g. Pandas, NumPy (there is a plugin for Nuitka specifically for NumPy), scikit-learn, etc. I definitely encourage you to pursue this path - it would be a very nice feature to have -, I just suspect that it's non-trivial to get working for non-trivial Python programs.